### DISCUSSION TITLE
How to drag a full group with subTargetCheck ?

### DISCUSSION THREAD

**Discussion by vshepel** [fabricjs/fabric.js#9786] [Q&A]
This works with ActiveSelection, but if I want to drag the entire group with subTargetCheck, it starts moving only the object under the cursor

https://github.com/fabricjs/fabric.js/assets/14239290/5e5ce08e-d4ec-44ae-9d5f-7f651588f72d

**Comment by asturur** [comment]
those are expectations on the default interactions that aren't met and will never be.
So what is your desired ux?
It looks like you both have subTargetCheck and interactive set as true on the group. But you would like sometimes the group to behave as a single object?

**Reply by vshepel** [reply]
Yes, you right. I just need that when dragging an object in a group, the whole group would be dragged as in the case of ActiveSelection.

**Reply by vshepel** [reply]
With subTargetCheck and interactive - true, yes

**Reply by asturur** [reply]
So what you would like to happen?
how do you drag a single object inside a group? how you differentiate dragging a single object vs dragging the whole group?

**Reply by vshepel** [reply]
I want to move only the full group without moving the object inside it

See example, I can interact with items in a group, but if I need to move, then when I press the cursor on an item, the whole group is moved, not the individual item inside:

https://github.com/fabricjs/fabric.js/assets/14239290/e3f238c8-ee4b-4aae-951e-2d4a48c9c583

**Reply by jiayihu** [reply]
You need to play with the `interactive` flag. If you want to move the Group it should be false. Then as @asturur suggested, you need an UX that defines when the desired behaviour is moving the child vs moving the group. For instance, just as idea, if Shift key is pressed then you may want to disable `interactive` and allow moving the Group as whole.

**Reply by vshepel** [reply]
I don't need to move the child elements. I've also thought about enabling or disabling "interactive", but I don't know which event to put it on yet
